pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/worker/WorkerConfig.java

```diff
@@ -168,6 +168,11 @@
         doc = "The pulsar topic used for storing function metadata"
     )
     private String functionMetadataTopicName;
+    @FieldContext(
+            category = CATEGORY_FUNC_METADATA_MNG,
+            doc = "Should the metadata topic be compacted?"
+    )
+    private Boolean compactMetadataTopic = false;
```

Comment:
Given the current approach, this is a dangerous flag. If a user mistakenly flips the flag on.  It could render their cluster corrupt. We should perhaps add more warnings for what this config with do.  Instead of "compactMetadataTopic" maybe we can rename it to "useCompactedMetadataTopic"?  So that what it does is more clear.

Reply:
Will change the name. Although the possibility of corruption is low. Because before a leader can actually start writing, it needs to read all existing messages and the messages will not deserialize either way.

Reply:
Changed

Reply:
I would suggest to add some more content to the doc annotation on what the impact of flipping the flag is.